helix-rest/src/main/java/org/apache/helix/rest/server/resources/lock/LockAccessor.java

```diff
@@ -0,0 +1,228 @@
+        "Failed to lock the path %s using zk address %s, please check if this lock is held by someone else.",
+        lockLocator.getLockPath(), lockLocator.getZkAddress()));
+  }
+
+  private LockLocator parseLockLocator(ZNRecord znRecord) throws Exception {
+    if (znRecord == null) {
+      return null;
+    }
+
+    // Parse zk address from simple field
+    String zkAddr = znRecord.getSimpleField(ZK_ADDR);
+    if (zkAddr == null) {
+      throw new Exception("No ZK address is provided, could not create a lock.");
+    }
+
+    // Parse lock path from simple field
+    String lockPath = znRecord.getSimpleField(LOCK_PATH);
+
+    // If lock path is not provided, generate a lock path from lock scope
+    if (lockPath == null) {
+      Map<String, Map<String, String>> mapFields = znRecord.getMapFields();
+      LockScope lockScope = null;
+      for (LockScopeKey lockScopeKey : LockScopeKey.values()) {
+        // If a known lock scope is found, currently only HelixLockScope is available
+        if (mapFields.containsKey(lockScopeKey.name())) {
+          // Helix Lock Scope
+          if (lockScopeKey.name().equals(LockScopeKey.HelixLockScope.name())) {
+            lockScope = generateHelixLockScope(mapFields.get(lockScopeKey.name()));
+          }
+        }
+      }
+      if (lockScope == null) {
+        throw new Exception(
+            "Failed to parse a lock scope. Please check the lock scope property and path keys and make sure they are correct.");
+      }
+      return new LockLocator(zkAddr, lockScope.getPath());
+    }
+
+    return new LockLocator(zkAddr, lockPath);
+  }
+
+  private LockInfo parseLockInfo(ZNRecord znRecord) throws Exception {
+    if (znRecord == null) {
+      return null;
+    }
+    // Parse user iod
+    String userId = znRecord.getSimpleField(USER_ID);
+
+    // Parse lock timeout
+    long lockTimeout;
+    try {
+      lockTimeout = Long.parseLong(znRecord.getSimpleField(LOCK_TIMEOUT));
+    } catch (NumberFormatException e) {
+      throw new Exception("Please check the timeout value input, could not convert to long.");
+    }
+
+    // Parse lock message
+    String lockMsg = znRecord.getSimpleField(LOCK_MSG);
+    return new LockInfo(userId, lockMsg, lockTimeout);
+  }
+
+  private LockScope generateHelixLockScope(Map<String, String> lockScopeVal) throws Exception {
+    // Get the lock scope property: cluster, participant, resource, etc.
+    String CLUSTER = "cluster";
+    String RESOURCE = "resource";
+    String PARTICIPANT = "participant";
+    String lockScopeProperty = lockScopeVal.get(LOCK_SCOPE_PROPERTY);
+    HelixLockScope.LockScopeProperty property;
+    try {
+      property = HelixLockScope.LockScopeProperty.valueOf(lockScopeProperty);
+    } catch (IllegalArgumentException e) {
+      throw new Exception(String.format(
+          "User requested lock scope property %s, but this property does not exist in HelixLockScope.",
+          lockScopeProperty));
+    }
```

Comment:
I think there's a shorter way of doing this. Try googling? It looks like Enums.getIfPresent(Blah.class, "A")...